components/popover/popover-mixin.js

```diff
@@ -127,68 +215,375 @@ export const PopoverMixin = superclass => class extends superclass {
+			// removing the rAF call can allow infinite focus looping to happen in content using a focus trap
 			requestAnimationFrame(() => focusable.focus());
 		} else {
-			const content = this._getContentContainer();
+			const content = this.#getContentContainer();
 			content.setAttribute('tabindex', '-1');
 			content.focus();
 		}
 	}
 
-	_focusOpener() {
+	#focusOpener() {
 		if (!document.activeElement) return;
 		if (!isComposedAncestor(this, getComposedActiveElement())) return;
 
 		this?._opener.focus();
 	}
 
-	_getContentContainer() {
-		return this.shadowRoot.querySelector('.content');
+	#getContentContainer() {
+		return this.shadowRoot.querySelector('.content-container');
+	}
+
+	#getLocation(spaceAround, spaceAroundScroll, spaceRequired) {
+
+		const preferred = this._preferredPosition;
+		if (!preferred.allowFlip) {
+			return preferred.location;
+		}
+
+		if (preferred.location === 'block-end') {
+			if (spaceAround.below >= spaceRequired.height) return 'block-end';
+			if (spaceAround.above >= spaceRequired.height) return 'block-start';
+			// if auto-fit is enabled, scroll will be enabled for the inner content so it will always fit in the available space so pick the largest space it can be displayed in
+			if (!this.noAutoFit) return spaceAround.above > spaceAround.below ? 'block-start' : 'block-end';
+			if (spaceAroundScroll.below >= spaceRequired.height) return 'block-end';
+			if (spaceAroundScroll.above >= spaceRequired.height) return 'block-start';
+		}
+
+		if (preferred.location === 'block-start') {
+			if (spaceAround.above >= spaceRequired.height) return 'block-start';
+			if (spaceAround.below >= spaceRequired.height) return 'block-end';
+			// if auto-fit is enabled, scroll will be enabled for the inner content so it will always fit in the available space so pick the largest space it can be displayed in
+			if (!this.noAutoFit) return spaceAround.above > spaceAround.below ? 'block-start' : 'block-end';
+			if (spaceAroundScroll.above >= spaceRequired.height) return 'block-start';
+			if (spaceAroundScroll.below >= spaceRequired.height) return 'block-end';
+		}
+
+		// todo: add location order for inline-start and inline-end
+
+		// if auto-fit is disabled and it doesn't fit in the scrollable space above or below, always open down because it can add scrollable space
+		return 'block-end';
+	}
+
+	#getPointer() {
+		return this.shadowRoot.querySelector('.pointer');
+	}
+
+	#getPointerPosition(openerRect) {
+		const position = {};
+		const pointer = this.#getPointer();
+		if (!pointer) return position;
+
+		const pointerRect = pointer.getBoundingClientRect();
+		const isRTL = this.getAttribute('dir') === 'rtl';
+
+		if (this._preferredPosition.span !== 'all') {
+			const xAdjustment = Math.min(20 + ((pointerRotatedLength - pointerLength) / 2), (openerRect.width - pointerLength) / 2);
+			if (!isRTL) {
+				if (this._preferredPosition.span === 'end') {
+					position.left = openerRect.left + xAdjustment;
```

Comment:
If we used `inset-inline-start`/`inset-inline-end` instead, could we stop needing to know about `isRTL`?

Reply:
Haha, yeah. Added reply to your other comment. Really just comes down to browser support.

Reply:
Per offline discussion, I've updated the code to eliminate the usage of `RtlMixin`. Nothing was actually using `dir="rtl"` in CSS... so that wasn't an issue. 

However, there doesn't appear to be an API we can use to get "logical position", which is what we'd need to avoid checking RTL in the JavaScript. I think we can still simplify the calculations if we convert the opener's `getBoundingClientRect` result to logical values to be used throughout the rest of the positioning calcs and use `inset-inline-start|end` for positioning, but it requires a bit of an overhaul to those calcs. I think there is some value in preserving the current calcs in the history, so I'd like to do that separately.

I avoided duplicating what `RtlMixin` was doing to observe document language changes, which avoids a bunch of extra code which is only really needed for the case where the `dir` changes while the popover is open. Initially I had that, but then back-pedalled because it doesn't seem necessary.